src/app/shared/components/table/table.component.ts

```diff
@@ -0,0 +1,99 @@
+import { Component, EventEmitter, Input, OnChanges, Output, SimpleChanges, TemplateRef, ViewEncapsulation } from '@angular/core'
+import { Icon, IconSize } from '../../enums/icon'
+
+export enum SortDirection {
+  DESC = 'DESC',
+  ASC = 'ASC',
+}
+
+export interface SofieTableHeader<T> {
+  name: string
+  isBeingUsedForSorting?: boolean
+  sortCallback: (a: T, b: T) => number
+  sortDirection: SortDirection
+}
+
+@Component({
+  selector: 'sofie-table',
+  templateUrl: './table.component.html',
+  styleUrls: ['./table.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class TableComponent<T> implements OnChanges {
+  protected readonly IconSize = IconSize
+
+  @Input() public headers: SofieTableHeader<T>[] = []
+
+  @Input() public entities: T[] = []
+  @Input() public rowTemplate: TemplateRef<T>
+  @Input() public shouldShowRowCallback: (entity: T) => boolean = (): boolean => true
+
+  @Input() public showCheckboxes: boolean = false
+  @Input() public selectedEntities: Set<T> = new Set()
+  @Output() public selectedEntitiesChange: EventEmitter<Set<T>> = new EventEmitter()
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    if (changes['entities']) {
+      this.sortEntities()
+    }
+  }
+
+  public getRowTemplateContext(entity: T): { tableRowData: T } {
+    return {
+      tableRowData: entity,
+    }
+  }
+
+  public toggleTableHeaderForSorting(header: SofieTableHeader<T>): void {
+    header.isBeingUsedForSorting = true
+    header.sortDirection = header.sortDirection === SortDirection.DESC ? SortDirection.ASC : SortDirection.DESC
```

Comment:
Updating the object by references will modify state outside of the component.

Reply:
The alternative is to constantly recreate the entire headers array.
In this specific case the headers aren't supposed to be used outside this component. They only need to be provided.

Reply:
It is not up to the table to dictate how data are used outside of itself.
_constantly_ is an exaggeration, but yes, creating new objects for the column header and the header array with the modified state would be my suggestion.

Another alternative is to move the sort direction out of the header and have an attribute that holds the value of the header that is to be used for sorting. This will introduce two variables.

But it is fine if it is not part of this PR.